x-pack/plugins/apm/server/lib/services/get_service_dependencies/index.ts

```diff
@@ -198,32 +259,78 @@ export function getServiceDependencies({
+        (metric) =>
+          (metric.currentPeriod.latency.value ?? 0) *
+          (metric.currentPeriod.throughput.value ?? 0)
       )
       .filter(isFiniteNumber);
 
-    const minLatencySum = Math.min(...latencySums);
-    const maxLatencySum = Math.max(...latencySums);
+    const currentPeriodMinLatencySum = Math.min(...currentPeriodLatencySums);
+    const currentPeriodMaxLatencySum = Math.max(...currentPeriodLatencySums);
+
+    const previousPeriodLatencySums = metricsByResolvedAddress
+      .map(
+        (metric) =>
+          (metric.previousPeriod.latency.value ?? 0) *
+          (metric.previousPeriod.throughput.value ?? 0)
+      )
+      .filter(isFiniteNumber);
+
+    const previousPeriodMinLatencySum = Math.min(...previousPeriodLatencySums);
+    const previousPeriodMaxLatencySum = Math.max(...previousPeriodLatencySums);
 
     return metricsByResolvedAddress.map((metric) => {
-      const impact =
-        isFiniteNumber(metric.latency.value) &&
-        isFiniteNumber(metric.throughput.value)
-          ? ((metric.latency.value * metric.throughput.value - minLatencySum) /
-              (maxLatencySum - minLatencySum)) *
-            100
-          : 0;
+      const { currentPeriod, previousPeriod, ...rest } = metric;
+
+      const currentPeriodImpact = calculateImpact({
+        latency: currentPeriod.latency.value,
+        throughput: currentPeriod.throughput.value,
+        minLatency: currentPeriodMinLatencySum,
+        maxLatency: currentPeriodMaxLatencySum,
+      });
+
+      const previousPeriodImpact = calculateImpact({
+        latency: previousPeriod.latency.value,
+        throughput: previousPeriod.throughput.value,
+        minLatency: previousPeriodMinLatencySum,
+        maxLatency: previousPeriodMaxLatencySum,
+      });
 
       return {
-        ...metric,
-        impact,
+        ...rest,
+        currentPeriod: { ...currentPeriod, impact: currentPeriodImpact },
+        previousPeriod: {
+          latency: {
+            ...previousPeriod.latency,
+            timeseries: offsetPreviousPeriodCoordinates({
+              currentPeriodTimeseries: currentPeriod.latency.timeseries,
+              previousPeriodTimeseries: previousPeriod.latency.timeseries,
+            }),
+          },
+          throughput: {
+            ...previousPeriod.throughput,
+            timeseries: offsetPreviousPeriodCoordinates({
+              currentPeriodTimeseries: currentPeriod.throughput.timeseries,
+              previousPeriodTimeseries: previousPeriod.throughput.timeseries,
+            }),
+          },
+          errorRate: {
+            ...previousPeriod.errorRate,
+            timeseries: offsetPreviousPeriodCoordinates({
+              currentPeriodTimeseries: currentPeriod.errorRate.timeseries,
+              previousPeriodTimeseries: previousPeriod.errorRate.timeseries,
+            }),
+          },
+          impact: previousPeriodImpact,
+        },
```

Comment:
Also some opportunity to reduce repeated blocks of code:

```typescript
const metricKeys = ['latency', 'throughput', 'errorRate'] as const;
const offSetPreviousPeriod = metricKeys.reduce(
  (acc, key) => ({
    ...acc,
    [key]: {
      ...previousPeriod[key],
      timeseries: offsetPreviousPeriodCoordinates({
        currentPeriodTimeseries: currentPeriod[key].timeseries,
        previousPeriodTimeseries: previousPeriod[key].timeseries,
      }),
    },
  }),
  {} as Metrics
);
```

```suggestion
        previousPeriod: { ...offSetPreviousPeriod, impact: previousPeriodImpact },
```